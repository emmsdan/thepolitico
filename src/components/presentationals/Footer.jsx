import React from 'react';
import PropTypes from 'prop-types';

const Footer = props => {
  return (
    <footer>
      <div>
        <a href={props.footer.url}>Politico</a> powered by
        <a href={props.footer.poweredby.url}> {props.footer.poweredby.text} </a>
      </div>
    </footer>
  );
};

Footer.propTypes = {
  footer: PropTypes.shape({
    url: PropTypes.string,
    poweredby: PropTypes.objectOf({
      url: PropTypes.string,
      text: PropTypes.string,
    }),
  }),
};

export default Footer;
