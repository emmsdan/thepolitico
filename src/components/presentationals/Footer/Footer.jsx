import React from 'react';
import PropTypes from 'prop-types';

const Footer = ({ url, poweredby }) => {
  return (
    <footer>
      <div>
        <a href={url}>Politico</a> powered by
        <a href={poweredby.url}> {poweredby.name} </a>
      </div>
    </footer>
  );
};

Footer.propTypes = {
  url: PropTypes.string,
  poweredby: PropTypes.shape({
    url: PropTypes.string,
    name: PropTypes.string,
  }),
};

export default Footer;
