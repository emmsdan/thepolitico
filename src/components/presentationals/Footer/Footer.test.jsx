import React from 'react';
import { shallow } from 'enzyme';

import Footer from './Footer';

describe('<Footer />', () => {
  const footer = {
    url: 'http://thepolitico.io/about',
    poweredby: {
      url: 'http://twitter.com/emmsdan',
      name: 'emmanuel daniel',
    },
  };

  it('should render correctly', () => {
    const component = shallow(<Footer {...footer} />);
    expect(component).toMatchSnapshot();
  });
});
