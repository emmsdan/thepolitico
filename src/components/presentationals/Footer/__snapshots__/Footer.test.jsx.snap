// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> should render correctly 1`] = `
<footer>
  <div>
    <a
      href="http://thepolitico.io/about"
    >
      Politico
    </a>
     powered by
    <a
      href="http://twitter.com/emmsdan"
    >
       
      emmanuel daniel
       
    </a>
  </div>
</footer>
`;
